### DISCUSSION TITLE
Losing custom formatting for list migration

### DISCUSSION THREAD

**Discussion by harveryang** [Zerg00s/FlowPowerAppsMigrator#74] [Q&A] [answered]
This tool is undeniably effective for migrating PowerApps and Flows associated with lists. However, I observed that when a list incorporates custom formatting—such as a custom style applied to a column or view—the custom style JSON loses its original format, resulting in a disruption to the custom formatting. Is this a known issue? Anyway to workaround it?

**Comment by Zerg00s** [comment]
Hi @harveryang. Yes, I saw this problem before. Let me have a look at it.

**Reply by harveryang** [reply]
Thank you! Let me know if you need help to test the fix out.

**Reply by Zerg00s** [reply]
I might need some help, yes. Can you, please, share the Lists.xml file with me? I want to have a look at the formatting that you have that is not working.

**Reply by harveryang** [reply]
[Lists.zip](https://github.com/Zerg00s/FlowPowerAppsMigrator/files/13982395/Lists.zip)
Attached the Lists.xml file (in Zip). Please check the pnp:Fields node especially the CustomFormatter attributes at Field nodes.

**Reply by Zerg00s** [reply]
Hi @harveryang. I did more tests and it seems that the older version of the PnP PowerShell module (SharePointPnPPowerShellOnline) "does not know" how to import the CustomFormatter attribute correctly.

The new PnP.PowerShell module does not have this issue. I need to think about it.

**Reply by harveryang** [reply]
@Zerg00s While upgrading to the new PnP.PowerShell model may introduce additional issues or bugs to the solution, it is essential for the tool's continued improvement and functionality.

**Reply by Zerg00s** [reply, marked as answer]
You are right. The only issue is that the PnP.PowerShell module won't work without the Azure App registration or at least the SharePoint-only app. Most users are not M365 Tenant Admins or even SharePoint Admins. This is the reason I chose to use the legacy module. The legacy module works with the good old browser-based authentication, with a pop-up window. I really need to think about what to do next. So far, if the column formatting is the only issue, it's not worth the upgrade yet.
